src/nvim/ascii.h

```diff
@@ -146,7 +146,7 @@ static inline bool ascii_isxdigit(int c)
 /// Checks if `c` is an “identifier” character
 ///
 /// That is, whether it is alphanumeric character or underscore.
-static inline bool ascii_isident(const int c)
+static inline bool ascii_isident(int c)
```

Comment:
I agree with @aktau that `const` on parameters is usually more trouble than it's worth, because it cascades to callers. If the function body is large then a local `const` variant could be used instead.

Reply:
I have const in parameters for the similar reasons languages like Rust have const by default: it is safer to have data immutable by default. Wondering why MSVC has any troubles here, function definitely does not change `c`.

Reply:
I should really update declarations generator, number of `static inline` functions is growing.

Reply:
Right. `const` on values restricts the callee while telling the caller nothing. In contrast, `const` on "references" (like a pointer) does tell the caller something (I won't do anything to the value your pointer points to).  In the same vein, `restrict` tells the caller something, and is thus also useful in prototypes (thought only in very very select circumstances). 

At that point, my argument for not declaring parameters like `const size_t ...` is just to avoid word soup and to let callers focus on what's important.

In (really) long functions, I prefer not modifying variables/parameters: I'd declare a new one with a better name. 

In short(er) functions, I find that modifying a parameter or variable usually makes the code even shorter. The modification is obvious as the function fits in a limited viewport. One can keep the value flow in one's head.

Another way of looking at it which is often tied to the length of the function: when you have to think hard about the new name of something, that usually means it's a good candidate for re-assignment instead of just creating a similar variable with a new name.

> I have const in parameters for the similar reasons languages like Rust have const by default: it is safer to have data immutable by default.

I'm not doubting that it restricts the actions of the programmer a bit, which is usually safer. I'd be interested to read about where this is safer in our codebase, given that we're talking about `const` on value types (`char, int, double, struct, ...`).